src/kg_gen/utils/llm_deduplicate.py

```diff
@@ -0,0 +1,305 @@
+        item_clusters = {}
+        plural_type = "entities" if type == "node" else "edges"
+        singular_type = "entity" if type == "node" else "edge"
+
+        self.logger.info(
+            "Starting deduplication of %s %s in cluster", len(cluster), plural_type)
+
+        processed_count = 0
+        while len(cluster) > 0:
+            processed_count += 1
+            item = cluster.pop()
+
+            self.logger.debug(
+                "[%s/%s] Processing %s: '%s'", processed_count, len(cluster), singular_type, item)
+
+            relevant_items = self.get_relevant_items(item, 16, type)
+
+            self.logger.debug(
+                "  Found %s relevant %s for '%s'", len(relevant_items), plural_type, item)
+            if len(relevant_items) > 0:
+                self.logger.debug("  Sample relevant items: %s%s", relevant_items[:3], ("..." if len(
+                    relevant_items) > 3 else ""))
+
+            class Deduplicate(dspy.Signature):
+                __doc__ = f"""Find duplicate {plural_type} for the item and an alias that best represents the duplicates. Duplicates are those that are the same in meaning, such as with variation in tense, plural form, stem form, case, abbreviation, shorthand. Return an empty list if there are none. 
+                """
+                item: str = dspy.InputField()
+                set: list[str] = dspy.InputField()
+                duplicates: list[str] = dspy.OutputField(
+                    description="Exact matches to items in {plural_type} set")
+                alias: str = dspy.OutputField(
+                    description=f"Best {singular_type} name to represent the duplicates, ideally from the {plural_type} set")
+
+            deduplicate = dspy.Predict(Deduplicate)
+            result = deduplicate(item=item, set=relevant_items)
+            items.add(result.alias)
+
+            # Filter duplicates to only include those that exist in the cluster
+            duplicates = [dup for dup in result.duplicates if dup in cluster]
+
+            if len(duplicates) > 0:
+                self.logger.debug(
+                    "  ✓ Found %s duplicates for '%s'", len(duplicates), item)
+                self.logger.info(
+                    "  → Using alias '%s' to represent: '%s' and %s", result.alias, item, duplicates)
+                item_clusters[result.alias] = {item}
+                for duplicate in duplicates:
+                    cluster.remove(duplicate)
+                    item_clusters[result.alias].add(duplicate)
+            else:
+                self.logger.debug(
+                    "  ✗ No duplicates found for '%s', keeping as is", item)
+                item_clusters[item] = {item}
+
+        self.logger.debug(
+            "Deduplication complete: %s unique %s from original %s", len(items), plural_type, processed_count)
+
+        return items, item_clusters
+
+    def deduplicate(self) -> Graph:
+        lm = dspy.LM(model="gemini/gemini-2.0-flash",
```

Comment:
@reinier-millo any chance you make this configurable? or pass a dspy.LM from somehwere else?

Reply:
Yes, for sure, It can be just remove, and use the dspy previous context. It was required to run in test script, but I'll adjust the test script to provide the proper context.

After removing it, the model and key will be configured directly on the modal when we set the params to generate new graph.